diff -ur xfce4-time-out-plugin-1.0.0-orig/configure.ac xfce4-time-out-plugin-1.0.0/configure.ac
--- xfce4-time-out-plugin-1.0.0-orig/configure.ac	2010-12-27 18:08:19.000000000 +0100
+++ xfce4-time-out-plugin-1.0.0/configure.ac	2011-03-01 13:42:57.007063515 +0100
@@ -72,7 +72,7 @@
 dnl *** Check for required packages ***
 dnl ***********************************
 XDT_CHECK_PACKAGE([GTK], [gtk+-2.0], [2.8.0])
-XDT_CHECK_PACKAGE([LIBXFCEGUI4], [libxfcegui4-1.0], [4.3.99.2])
+XDT_CHECK_PACKAGE([LIBXFCE4UI], [libxfce4ui-1], [4.8.0])
 XDT_CHECK_PACKAGE([LIBXFCE4UTIL], [libxfce4util-1.0], [4.3.99.2])
 XDT_CHECK_PACKAGE([LIBXFCE4PANEL], [libxfce4panel-1.0], [4.3.99.2])
 
diff -ur xfce4-time-out-plugin-1.0.0-orig/panel-plugin/Makefile.am xfce4-time-out-plugin-1.0.0/panel-plugin/Makefile.am
--- xfce4-time-out-plugin-1.0.0-orig/panel-plugin/Makefile.am	2010-12-06 17:34:50.000000000 +0100
+++ xfce4-time-out-plugin-1.0.0/panel-plugin/Makefile.am	2011-03-01 13:42:57.008063514 +0100
@@ -24,13 +24,13 @@
 	-DG_LOG_DOMAIN=\"xfce4-time-out-plugin\"			\
 	-DPACKAGE_LOCALE_DIR=\"$(localedir)\"				\
 	$(LIBXFCE4UTIL_CFLAGS)						\
-	$(LIBXFCEGUI4_CFLAGS)						\
+	$(LIBXFCE4UI_CFLAGS)						\
 	$(LIBXFCE4PANEL_CFLAGS)						\
 	$(PLATFORM_CFLAGS)
 
 xfce4_time_out_plugin_LDADD =						\
 	$(LIBXFCE4UTIL_LIBS)						\
-	$(LIBXFCEGUI4_LIBS)						\
+	$(LIBXFCE4UI_LIBS)						\
 	$(LIBXFCE4PANEL_LIBS)
 
 #
diff -ur xfce4-time-out-plugin-1.0.0-orig/panel-plugin/time-out.c xfce4-time-out-plugin-1.0.0/panel-plugin/time-out.c
--- xfce4-time-out-plugin-1.0.0-orig/panel-plugin/time-out.c	2010-12-06 17:34:50.000000000 +0100
+++ xfce4-time-out-plugin-1.0.0/panel-plugin/time-out.c	2011-03-01 14:05:27.385692514 +0100
@@ -25,9 +25,8 @@
 
 #include <gtk/gtk.h>
 #include <libxfce4util/libxfce4util.h>
-#include <libxfcegui4/libxfcegui4.h>
-#include <libxfce4panel/xfce-panel-plugin.h>
-#include <libxfce4panel/xfce-hvbox.h>
+#include <libxfce4ui/libxfce4ui.h>
+#include <libxfce4panel/libxfce4panel.h>
 
 #include "time-out.h"
 #include "time-out-countdown.h"
@@ -477,7 +476,7 @@
   gtk_window_set_position (GTK_WINDOW (dialog), GTK_WIN_POS_CENTER);
 
   /* Create time settings section */
-  framebox = xfce_create_framebox (_("Time settings"), &timebin);
+  framebox = xfce_gtk_frame_box_new (_("Time settings"), &timebin);
   gtk_container_set_border_width (GTK_CONTAINER (framebox), 6);
   gtk_box_pack_start (GTK_BOX (GTK_DIALOG (dialog)->vbox), framebox, TRUE, TRUE, 0);
   gtk_widget_show (framebox);
@@ -561,7 +560,7 @@
   gtk_widget_show (spin);
 
   /* Create behaviour section */
-  framebox = xfce_create_framebox (_("Behaviour"), &behaviourbin);
+  framebox = xfce_gtk_frame_box_new (_("Behaviour"), &behaviourbin);
   gtk_container_set_border_width (GTK_CONTAINER (framebox), 6);
   gtk_box_pack_start (GTK_BOX (GTK_DIALOG (dialog)->vbox), framebox, TRUE, TRUE, 0);
   gtk_widget_show (framebox);
@@ -586,7 +585,7 @@
   gtk_widget_show (checkbutton);
 
   /* Create appearance section */
-  framebox = xfce_create_framebox (_("Appearance"), &appearancebin);
+  framebox = xfce_gtk_frame_box_new (_("Appearance"), &appearancebin);
   gtk_container_set_border_width (GTK_CONTAINER (framebox), 6);
   gtk_box_pack_start (GTK_BOX (GTK_DIALOG (dialog)->vbox), framebox, TRUE, TRUE, 0);
   gtk_widget_show (framebox);
diff -ur xfce4-time-out-plugin-1.0.0-orig/panel-plugin/time-out-countdown.c xfce4-time-out-plugin-1.0.0/panel-plugin/time-out-countdown.c
--- xfce4-time-out-plugin-1.0.0-orig/panel-plugin/time-out-countdown.c	2010-12-11 15:32:22.000000000 +0100
+++ xfce4-time-out-plugin-1.0.0/panel-plugin/time-out-countdown.c	2011-03-01 13:42:57.013063514 +0100
@@ -23,7 +23,7 @@
 #include <config.h>
 #endif
 
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 #include "time-out-countdown.h"
 
 
diff -ur xfce4-time-out-plugin-1.0.0-orig/panel-plugin/time-out-fadeout.c xfce4-time-out-plugin-1.0.0/panel-plugin/time-out-fadeout.c
--- xfce4-time-out-plugin-1.0.0-orig/panel-plugin/time-out-fadeout.c	2010-12-06 17:34:50.000000000 +0100
+++ xfce4-time-out-plugin-1.0.0/panel-plugin/time-out-fadeout.c	2011-03-01 13:42:57.013063514 +0100
@@ -24,7 +24,7 @@
 #endif
 
 #include <gtk/gtk.h>
-#include <libxfce4panel/xfce-panel-plugin.h>
+#include <libxfce4panel/libxfce4panel.h>
 
 #include "time-out-fadeout.h"
 
diff -ur xfce4-time-out-plugin-1.0.0-orig/panel-plugin/time-out-lock-screen.c xfce4-time-out-plugin-1.0.0/panel-plugin/time-out-lock-screen.c
--- xfce4-time-out-plugin-1.0.0-orig/panel-plugin/time-out-lock-screen.c	2010-12-06 17:34:50.000000000 +0100
+++ xfce4-time-out-plugin-1.0.0/panel-plugin/time-out-lock-screen.c	2011-03-01 14:09:20.677280013 +0100
@@ -23,7 +23,7 @@
 #include <config.h>
 #endif
 
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 #include "time-out-lock-screen.h"
 #include "time-out-countdown.h"
@@ -272,10 +272,8 @@
   gdk_flush ();
 
   /* Determine screen on which the pointer currently is */
-  screen = xfce_gdk_display_locate_monitor_with_pointer (NULL, &monitor);
-
   /* Center window on target monitor */
-  xfce_gtk_window_center_on_monitor (GTK_WINDOW (lock_screen->window), screen, monitor);
+  xfce_gtk_window_center_on_active_screen (GTK_WINDOW (lock_screen->window));
 
   lock_screen->max_seconds = max_sec;
 
